Indicators/TomDemarkSequential.cs

```diff
@@ -0,0 +1,353 @@
+///   or greater than (Sell Setup) the close 4 bars earlier.
+/// - **TDST Support/Resistance**: After a valid 9-bar setup completes, the indicator records the **lowest low** (for Sell Setup)
+///   or **highest high** (for Buy Setup) among the 9 bars. These are referred to as Support Price(for Buy Setup) and
+///   Resistance Price (for Sell Setup), and serve as critical thresholds that validate the continuation of the countdown phase.
+/// - **Countdown Phase**: Once a valid setup is completed, the indicator attempts to count 13 qualifying bars (not necessarily consecutive)
+///   where the close is less than (Buy Countdown) or greater than (Sell Countdown) the low/high 2 bars earlier.
+///   During this phase, the TDST Support/Resistance levels are checked — if the price breaks these levels, the countdown phase is reset,
+///   as the trend reversal condition is considered invalidated.
+/// </summary>
+/// <remarks>
+/// This implementation uses a rolling window of 10 bars and begins logic when at least 5 bars are available.
+/// </remarks>
+/// <seealso cref="IBaseDataBar"/>
+/// <seealso href="https://demark.com/sequential-indicator/"/>
+/// <seealso href="https://practicaltechnicalanalysis.blogspot.com/2013/01/tom-demark-sequential.html"/>
+/// <seealso href="https://medium.com/traderlands-blog/tds-td-sequential-indicator-2023-f8675bc5d14"/>
+
+public class TomDemarkSequential : WindowIndicator<IBaseDataBar>
+{
+    /// <summary>
+    /// The fixed number of bars used in the setup phase, as defined by the Tom DeMark Sequential (TD Sequential) indicator.
+    /// According to the TomDemark Sequential rules, a setup consists of exactly 9 consecutive qualifying bars.
+    /// </summary>
+    private const int MaxSetupCount = 9;
+
+    /// <summary>
+    /// The fixed number of bars used in the countdown phase, per the Tom Demark Sequential methodology.
+    /// A countdown is completed after exactly 13 qualifying bars.
+    /// These values are not configurable as they are fundamental to the algorithm's definition.
+    /// </summary>
+    private const int MaxCountdownCount = 13;
+    
+    /// <summary>
+    /// The windows of size 10 bars are enough and begin logic when at least 5 bars are available.
+    /// </summary>
+    private const int WindowSize = 10;
+
+    /// <summary>
+    ///  The TomDemark Sequential setup logic requires at least 5 bars to begin evaluating valid setups.
+    /// </summary>
+    private const int RequiredSamples = 5;
+
+    private static decimal Default => EncodeState(TomDemarkSequentialPhase.None, 0);
+
+    private int _stepCount;
+
+    private TomDemarkSequentialPhase _currentPhase;
+    
+    private decimal _resistancePrice; // highest high of the 9-bar Tom Demark Sequential buy setup (indicates resistance)
+    private decimal _supportPrice; // lowest low of the 9-bar Tom Demark Sequential sell setup (indicates support)
```

Comment:
Let's make these a public readonly property, they seems interesting price points to be consumed 👍